torchgeo/datasets/usavars.py

```diff
@@ -0,0 +1,227 @@
+    - tree cover
+    - elevation
+    - population density
+    - nighttime lights
+    - income per houshold
+    - road length
+    - housing price
+
+    .. versionadded:: 0.3
+    """
+
+    url_prefix = (
+        "https://files.codeocean.com/files/verified/"
+        + "fa908bbc-11f9-4421-8bd3-72a4bf00427f_v2.0/data/int/applications"
+    )
+    pop_csv_suffix = "CONTUS_16_640_POP_100000_0.csv?download"
+
+    data_url = "https://mosaiks.blob.core.windows.net/datasets/uar.zip"
+    dirname = "usavars"
+    zipfile = "uar.zip"
+
+    md5 = "677e89fd20e5dd0fe4d29b61827c2456"
+
+    label_urls = {
+        "housing": f"{url_prefix}/housing/outcomes_sampled_housing_{pop_csv_suffix}",
+        "income": f"{url_prefix}/income/outcomes_sampled_income_{pop_csv_suffix}",
+        "roads": f"{url_prefix}/roads/outcomes_sampled_roads_{pop_csv_suffix}",
+        "nightligths": f"{url_prefix}/nightlights/outcomes_sampled_nightlights_{pop_csv_suffix}",
+        "population": f"{url_prefix}/population/outcomes_sampled_population_{pop_csv_suffix}",
+        "elevation": f"{url_prefix}/elevation/outcomes_sampled_elevation_{pop_csv_suffix}",
+        "treecover": f"{url_prefix}/treecover/outcomes_sampled_treecover_{pop_csv_suffix}",
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new USAVars dataset instance."""
+        self.root = root
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        self.files = self._load_files()
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample = self.files[index]
+        tensor_sample = {}
+        tensor_sample["image"] = self._load_image(sample["image"])
+
+        keys = [key for key in sample.keys() if key != "image"]
+        for key in keys:
+            tensor_sample[key] = Tensor([sample[key]])
+
+        if self.transforms is not None:
+            tensor_sample = self.transforms(tensor_sample)
+
+        return tensor_sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self) -> List[Dict[str, Any]]:
+        import pandas as pd
+
+        file_path = os.path.join(self.root, self.dirname, "uar")
+        files = os.listdir(file_path)
```

Comment:
You do not need to enumerate the directory if you already know the ids of all the files that are expected to be in there (which you do because you have the IDs in the CSV files right?)

Reply:
This should speed everything up significantly

Reply:
Yeah so we can just load the csv's in __init__ and only get the labels whenever we __get__ item right?

Reply:
I mean you can create the id --> fn mapping without doing the os.listdir(...) by assuming that the tiffs in `uar/` exist